### PR TITLE
add a wait condition in SendQueue if there's nothing to do

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: libraries/networking/src/udt/SendQueue.h [hunk 1]
```diff
@@ -79,17 +96,17 @@ private slots:
...
     
-    mutable QReadWriteLock _packetsLock; // Protects the packets to be sent list.
+    mutable std::mutex _packetsLock; // Protects the packets to be sent list.
     std::list<std::unique_ptr<Packet>> _packets; // List of packets to be sent
     
     Socket* _socket { nullptr }; // Socket to send packet on
     HifiSockAddr _destination; // Destination addr
     
-    std::atomic<uint32_t> _lastACKSequenceNumber; // Last ACKed sequence number
+    std::atomic<uint32_t> _lastACKSequenceNumber { 0 }; // Last ACKed sequence number
```
Comment (birarda) [inline libraries/networking/src/udt/SendQueue.h:105]: surprised this didn't initialize to 0 automatically

**Comment by birarda** [comment]
Once @Atlante45's cleanup code is in it's possible we'd want to scrap this approach all together. It's conceivable that we'd lock the send queue for a long time in that `std::lock` while it's getting both the packets and loss list lock - we'd have to do some testing at capacity to see how much time we spend there.

File: libraries/networking/src/udt/SendQueue.h (hunk 1 above)
Comment (Atlante45) [inline libraries/networking/src/udt/SendQueue.h:105]: Must be acting like integer types.

Comment (birarda) [inline libraries/networking/src/udt/SendQueue.h:105]: Yep, I am dumb. I was reading `std::atomic` docs and read that zero initialization is performed by default. For `int` values (or `uint32_t`) that doesn't mean that it is set to zero.

**Pull request merged**